hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -250,7 +250,7 @@ public void burn(
 			final var uniqueToken = loadedUniqueTokens.get(serialNum);
 			validateTrue(uniqueToken != null, FAIL_INVALID);
 
-			final var treasuryIsOwner = uniqueToken.getOwner().equals(Id.DEFAULT);
+			final var treasuryIsOwner = uniqueToken.getOwner().equals(treasuryId);
```

Comment:
Hmm...when the treasury owns an NFT, the owner _is_ supposed to be the "wildcard" `0.0.0` id.

This way, a `TokenUpdate` can change a token's treasury without needing to iterate through all the NFTs and updating their `owner` fields. 

With this change, EETs like [this one](https://github.com/hashgraph/hedera-services/blob/master/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java#L744) will break.

Reply:
The use case is trying to burn an NFT that was sent out.  mint, transfer to user, user transfers back to treasury, then burn.

It looks like the owner of a trasury NFT needs to be 0.0.0, so do we need to add a check on an NFT transfer that if the new owner is the treasury the owner gets set to 0.0.0 instead of the treasury account.